### ISSUE TITLE
Add `lemonade bench` command

### ISSUE THREAD

**Issue by bitgamma** [lemonade-sdk/lemonade#1943]
Add a `benchmark` feature to the lemonade CLI.

The main initial goal is to provide speed metrics for installed models using different backends in an easy way. The tools will use regular chat completion endpoint so we don't measure raw inference speed, rather e2e performance.

The benchmark must play scripts that measure performance with different scenarios (relevant with techniques like MTP where coding gets a larger speed gain than, say, creative writing) and different context sizes. While the tool will need to offer some parameters, OOB ideally you would just need to type `lemonade bench modelname` to get some meaningful numbers.

Details TBD

**Comment by amd-pworfolk** [comment]
Make sure to take a look at [lemonade-eval](https://github.com/lemonade-sdk/lemonade-eval).  I'd be happy to see this functionality built into the core `lemonade` tool.

**Comment by Jays-1111** [comment]
Does this tool support benchmarking local custom models?

For example, can it evaluate the accuracy of your own fine-tuned and quantized Ryzen AI NPU/hybrid models across different datasets and report benchmark scores?

**Comment by amd-pworfolk** [comment]
> Does this tool support benchmarking local custom models?
> 
> For example, can it evaluate the accuracy of your own fine-tuned and quantized Ryzen AI NPU/hybrid models across different datasets and report benchmark scores?

@Jays-1111: [lemonade-eval#accuracy-testing](https://github.com/lemonade-sdk/lemonade-eval#accuracy-testing) has some of these capabilities (perplexity, ...).  'lemonade bench' does TTFT and TPS measurements.

**Comment by Jays-1111** [comment]
> > Does this tool support benchmarking local custom models?
> > For example, can it evaluate the accuracy of your own fine-tuned and quantized Ryzen AI NPU/hybrid models across different datasets and report benchmark scores?
> 
> [@Jays-1111](https://github.com/Jays-1111): [lemonade-eval#accuracy-testing](https://github.com/lemonade-sdk/lemonade-eval#accuracy-testing) has some of these capabilities (perplexity, ...). 'lemonade bench' does TTFT and TPS measurements.

@amd-pworfolk Thanks, this tool is great, I've used it before, but I encountered some issues regarding the evaluation of the npu model. If possible, please take a look. Issue: https://github.com/lemonade-sdk/lemonade-eval/issues/8 

@bitgamma I mainly want this tool to support loading models from local dir.

**Comment by amd-pworfolk** [comment]
Ahh, yes, after the `lemonade-server` refactor, I forgot that the accuracy tests in `lemonade-eval` are broken!  The logits are no longer available to compute perplexity and there are different issues with the other tests!  This really needs a total revamp as only the benchmarking (TTFT and TPS) and power and memory profiling are working.  With benchmarking going into `lemonade-server`, we need to revisit the value proposition for `lemonade-eval`!

**Comment by bitgamma** [comment]
The tool has been added with #1953
